### ISSUE TITLE
Replay associated error timestamp

### ISSUE THREAD

**Issue by vaind** [getsentry/sentry#71890]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

I've noticed this when working on RN replay touch breadcrumbs:

When an error is trigerred by a buton press, the error shows up before the actual touch event even though it should be the other way around. 
The issue is caused by the error not having an exact sub-second timestmap. However, it seems the timestmap is set correctly (with millisecond precision) when the event is sent from the javascript SDK, but then is trimmed to second-precision.

### Expected Result

Breadcrumbs list touch events before the error caused by the touch event.

### Actual Result

Example (the last listed touch event is the one that caused the error):
<img width="1154" alt="image" src="https://github.com/getsentry/sentry/assets/6349682/e8bc8974-ef62-416c-aef0-2f4b4ce96be1">

### Product Area

Unknown

### Link

https://sentry-sdks.sentry.io/replays/7eaabdb7c8374de292e9dbb841d9b777/?project=5428561&query=&referrer=%2Freplays%2F&statsPeriod=24h&t_main=breadcrumbs&yAxis=count%28%29

**Comment by cmanallen** [comment]
@vaind I think its a data-type issue.  Type 3 is an int and type 5 is a float.

JS SDK:
<img width="344" alt="Screenshot 2024-06-03 at 9 05 29 AM" src="https://github.com/getsentry/sentry/assets/6018782/7778ccd4-b955-4145-aac3-2efe0f80f063">


RN SDK:
<img width="335" alt="Screenshot 2024-06-03 at 9 55 32 AM" src="https://github.com/getsentry/sentry/assets/6018782/b73cdf7e-8fb7-413e-9766-ebcf64f46690">

**Comment by vaind** [comment]
> I think its a data-type issue. Type 3 is an int and type 5 is a float.

Where exactly are you seeing these?

**Comment by cmanallen** [comment]
@vaind In the network tab.  The request ending in `/recording-segments/` shows RN SDK type 5 events with a integer timestamp.  I assume there's some issue with the way the client parses those ints that's causing your ordering issues.  But I'm not sure.

cc @ryan953 @michellewzhang

**Comment by michellewzhang** [comment]
cc @billyvg

**Comment by michellewzhang** [comment]
@vaind where are you seeing the events being trimmed to second-precision?

**Comment by vaind** [comment]
> @vaind where are you seeing the events being trimmed to second-precision?

In the error JSON as shown in the Issues section (https://us.sentry.io/api/0/projects/sentry-sdks/sentry-react-native/events/998a79f1f18341b7b9d5747ff38f650e/json/). But it may not be relevant to ordering in replay, considering what @cmanallen pointed out.

**Comment by cmanallen** [comment]
@ryan953 Why is this just a problem for the RN SDK and not the Javascript SDK?

**Comment by billyvg** [comment]
@cmanallen it's been a problem for JS SDK FWIW, but it's possible it's an SDK problem there.

**Comment by aliu39** [comment]
@ryan953 
> The endpoint to check is /replays-events-meta/ and not /events/

From what I looked at, it seems like the replay endpoint queries Discover under the hood, same as /events. It doesn't do anything to format/truncate the timestamp field.

From the examples [in sdk docs](https://develop.sentry.dev/sdk/event-payloads/types/), it looks like we support milliseconds, but it's up to the SDK to send them.

**Comment by vaind** [comment]
AFAICT the timestmap has the millisecond precision when the event is sent from the SDK.

**Comment by cmanallen** [comment]
Timestamps are stored to second precision.  There's nothing we can do on the back-end.  The web client will need to special case this to fix it.

**Comment by vaind** [comment]
To me the other of breadcrumbs is pretty important because it implies causality, i.e. that the error was caused by the click. As it is, the error seem to happen before user action, which is incorrect.

What can be done to make this work?

**Comment by bruno-garcia** [comment]
Lets follow up sometime next week once @aliu39 is back from pto

**Comment by aliu39** [comment]
The team will be looking into ways to fill in the missing precision this week

**Comment by aliu39** [comment]
Related: https://www.notion.so/sentry/DACI-Increase-the-precision-of-error-events-timestamp-e14319f686bb498997c7eee42c35bed8

**Comment by aliu39** [comment]
Unfortunately, until the initiative linked above is carried out, we're not able to support this.

**Comment by aliu39** [comment]
[New notion page](https://www.notion.so/sentry/Increase-Error-Event-Timestamp-Precision-2ecc1f8f1361488b9297013aa5ae284c?pvs=4)

**Comment by billyvg** [comment]
Related https://github.com/getsentry/sentry/issues/68752

**Comment by cmanallen** [comment]
This will be fixed if/when the errors platform records millisecond precision on their timestamps.  In the meantime we could emit a breadcrumb on the replay with this information and have our UI fix the precision issue.  If the SDK team does not want to do this then we should close the issue and track on Armen's errors issue.

**Comment by yuvmen** [comment]
@chargome I see you moved this to TODO, you should know we are working on correcting this on the errors platform and plan on resolving this soon (~week) on the backend.

**Comment by aliu39** [comment]
Closed by the PR above, ms are showing up in UI!

**Comment by aliu39** [comment]
@yuvmen mentioned this issue has only been fixed for **error events**, not non-error issue platform events, ex "Slow DB query", and I'm guessing UF

**Comment by aliu39** [comment]
@cvxluo you recently closed this out right? Is there another GH or linear ticket you can link

**Comment by cvxluo** [comment]
yes — here's the project: https://linear.app/getsentry/project/increase-error-event-timestamp-precision-2b0bd78e0f3d/overview

the ticket that closes this: https://linear.app/getsentry/issue/ID-605/precise-error-ordering-on-replay-breadcrumbs-queries

**Comment by aliu39** [comment]
Great thanks!!
